### PR TITLE
Add metrics to track channels

### PR DESCRIPTION
**Pull request by pvdrz**
cc @Alexandcoats

### CODE REVIEW

File: bee-metrics/Cargo.toml [hunk 1]
```diff
@@ -15,6 +15,12 @@ axum = { version = "0.4.8", default-features = false, features = [ "http2" ] }
 parking_lot = { version = "0.12.0", default-features = false }
 prometheus-client = { version = "0.15.1", default-features = false }
 tokio = { version = "1.17.0", default-features = false, features = [ "process" ] }
+tokio-stream = { version = "0.1.8", default-features = false, optional = true }
+futures-core = { version = "0.3.21", default-features = false, optional = true }
```
Comment (thibault-martinez) [inline bee-metrics/Cargo.toml]: Order!

File: bee-metrics/src/metrics/sync/mpsc.rs [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    }
+
+    #[inline]
+    fn metric_type(&self) -> MetricType {
+        self.0.metric_type()
+    }
+}
+
+/// A wrapper around [`UnboundedReceiver`] that implements [`Stream`].
+pub struct UnboundedReceiverStream<T>(TokioUnboundedReceiverStream<T>, Counter);
```
Comment (grtlr) [inline bee-metrics/src/metrics/sync/mpsc.rs]: [nitpick] I think I would use named fields here.

**Review by grtlr — DISMISSED** [review]

File: bee-metrics/src/metrics/mod.rs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Module containing some predefined useful metrics.
```
Comment (thibault-martinez) [inline bee-metrics/src/metrics/mod.rs:4]:
```suggestion
//! Module containing some predefined useful metrics.

```

File: bee-metrics/src/metrics/sync/mpsc.rs [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+}
+
+/// A wrapper around [`UnboundedReceiver`] that implements [`Stream`].
+pub struct UnboundedReceiverStream<T> {
+    stream: TokioUnboundedReceiverStream<T>,
+    counter: Counter,
+}
+
+impl<T> UnboundedReceiverStream<T> {
+    /// Create a new [`UnboundedReceiverStream`].
```
Comment (thibault-martinez) [inline bee-metrics/src/metrics/sync/mpsc.rs]:
```suggestion
    /// Creates a new [`UnboundedReceiverStream`].
```

File: bee-metrics/src/metrics/sync/mpsc.rs [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+    #[inline]
+    pub fn new(recv: UnboundedReceiver<T>) -> Self {
+        Self {
+            stream: TokioUnboundedReceiverStream::new(recv.receiver),
+            counter: recv.counter,
+        }
+    }
+
+    /// Get back the inner [`UnboundedReceiver`].
```
Comment (thibault-martinez) [inline bee-metrics/src/metrics/sync/mpsc.rs]:
```suggestion
    /// Gets back the inner [`UnboundedReceiver`].
```

**Review by thibault-martinez — COMMENTED** [review]
Can you add entries in the changelog ? Do you want to bump the version ? I see that the previous version wasn't released so maybe include in 0.1 and change the date ?

**Review by thibault-martinez — APPROVED** [review]

**Review by grtlr — APPROVED** [review]

**Pull request merged**